### ISSUE TITLE
Incompatibility with 'Luna Slimes'

### ISSUE THREAD

**Issue by Mirsario** [PinkGoosik/visuality#37]
https://modrinth.com/mod/luna-slimes

This is an issue I should probably be opening on the other mod's issue tracker (as that mod is more recent), but to my knowledge it has no tracker nor privately-messageable contact info, so excuse me while I use this as a place to ping @AViewFromTheTop (aka Lunade_).

## The issue

Running `Visuality` with `LunaSlimes` (and nothing else besides for dependencies) causes a load-time crash due to conflicting mixins.

If it matters, and I don't think it does - I'm using Quilt, and QSL in place of Fabric API.

## Logs
**Log file:** [latest.log](https://github.com/PinkGoosik/visuality/files/10316537/latest.log)

**Relevant portion:**
```
[00:17:10] [main/WARN]: @Redirect conflict. Skipping lunaslimes.mixins.json:SlimeMixin from mod lunaslimes->@Redirect::stopParticles(Lnet/minecraft/class_1937;Lnet/minecraft/class_2394;DDDDDD)V with priority 1000, already redirected by visuality.mixins.json:SlimeEntityMixin from mod visuality->@Redirect::addParticle(Lnet/minecraft/class_1937;Lnet/minecraft/class_2394;DDDDDD)V with priority 1000
[00:17:10] [main/WARN]: Mixin transformation of net.minecraft.class_1621 failed
org.spongepowered.asm.mixin.transformer.throwables.MixinTransformerError: An unexpected critical error was encountered
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:392) ~[sponge-mixin-0.11.4+mixin.0.8.5.jar:0.11.4+mixin.0.8.5]
```

**Comment by Mirsario** [comment]
@AViewFromTheTop (Is the ping even working if I'm still the only participant?)
The 1.3 version of Luna Slimes fixed the startup crash, but now the conflict occurs in ticking of slimes:

[crash-2022-12-29_16.21.33-server.txt](https://github.com/PinkGoosik/visuality/files/10319666/crash-2022-12-29_16.21.33-server.txt)
```
java.lang.ClassCastException: class java.lang.Double cannot be cast to class net.minecraft.class_2394 (java.lang.Double is in module java.base of loader 'bootstrap'; net.minecraft.class_2394 is in unnamed module of loader org.quiltmc.loader.impl.launch.knot.KnotClassLoader @626c44e7)
	at org.spongepowered.asm.synthetic.args.Args$1.set(Args$1.java)
	at net.minecraft.entity.mob.SlimeEntity.args$zcj000$stopParticles(SlimeEntity:1202)
	at net.minecraft.entity.mob.SlimeEntity.tick(SlimeEntity:145)
```
`class_2394` is `net/minecraft/particle/ParticleEffect`

**Comment by AViewFromTheTop** [comment]
> @AViewFromTheTop (Is the ping even working if I'm still the only participant?) The 1.3 version of Luna Slimes fixed the startup crash, but now the conflict occurs in ticking of slimes:
> 
> [crash-2022-12-29_16.21.33-server.txt](https://github.com/PinkGoosik/visuality/files/10319666/crash-2022-12-29_16.21.33-server.txt)
> 
> ```
> java.lang.ClassCastException: class java.lang.Double cannot be cast to class net.minecraft.class_2394 (java.lang.Double is in module java.base of loader 'bootstrap'; net.minecraft.class_2394 is in unnamed module of loader org.quiltmc.loader.impl.launch.knot.KnotClassLoader @626c44e7)
> 	at org.spongepowered.asm.synthetic.args.Args$1.set(Args$1.java)
> 	at net.minecraft.entity.mob.SlimeEntity.args$zcj000$stopParticles(SlimeEntity:1202)
> 	at net.minecraft.entity.mob.SlimeEntity.tick(SlimeEntity:145)
> ```
> 
> `class_2394` is `net/minecraft/particle/ParticleEffect`

AAAAAAAAAAAAAAAAAAAAA

**Comment by AViewFromTheTop** [comment]
gonna try to release 1.4 soon
that's definitely an issue with compiling so i'm flat-out casting the doubles to doubles this time

**Comment by AViewFromTheTop** [comment]
Just looked into it, that crash shouldn't be happening to begin with. ParticleOptions (class_2394) is the first parameter, which would use 0 and not 1 for ModifyArgs. I'm not sure what I can really do there, that's not something I'm causing.
Edit: Just released 1.4. May take a bit to show up on CurseForge, but it's on Modrinth. I... might've fixed it..? I've never gotten that error, so I can't really tell.

**Comment by Mirsario** [comment]
> Just released 1.4. May take a bit to show up on CurseForge, but it's on Modrinth. I... might've fixed it..? I've never gotten that error, so I can't really tell.

It's working now, and slimes are looking really nice, thanks!
But please, make a modrinth-linked repo for your mod later, even if it'll be an empty one just for the issue tracker, as it took me a bit of digging to figure out how to report this to you.

**Comment by AViewFromTheTop** [comment]
> It's working now, and slimes are looking really nice, thanks! But please, make a modrinth-linked repo for your mod later, even if it'll be an empty one just for the issue tracker, as it took me a bit of digging to figure out how to report this to you.

Understood! The git repo is actually just private at the moment, so I'll just make it public. Thank you!!
